File: common/src/main/java/generations/gg/generations/core/generationscore/world/item/legends/SingleElmentPostUpdatingItem.java
package generations.gg.generations.core.generationscore.world.item.legends;

import com.cobblemon.mod.common.api.types.ElementalType;
import com.cobblemon.mod.common.battles.actor.PlayerBattleActor;
import com.google.common.collect.Streams;
import generations.gg.generations.core.generationscore.world.item.ItemWithLangTooltipImpl;
import generations.gg.generations.core.generationscore.world.item.LangTooltip;
import generations.gg.generations.core.generationscore.world.item.PostBattleUpdatingItem;
import net.minecraft.world.item.ItemStack;

public class SingleElmentPostUpdatingItem extends ItemWithLangTooltipImpl implements PostBattleUpdatingItem, LangTooltip {

    private final ElementalType type;

    public SingleElmentPostUpdatingItem(Properties properties, ElementalType type) {
        super(properties);
        this.type = type;
    }

    @Override
    public boolean checkData(PlayerBattleActor player, ItemStack stack, BattleData pixelmonData) {
         return Streams.stream(pixelmonData.pokemon().getTypes()).anyMatch(elementalType -> elementalType.equals(type));
    }
}

File: src/common/main/kotlin/generations/gg/generations/core/generationscore/common/world/item/legends/SingleElmentPostUpdatingItem.kt
package generations.gg.generations.core.generationscore.common.world.item.legends

import com.cobblemon.mod.common.api.types.ElementalType
import com.cobblemon.mod.common.battles.actor.PlayerBattleActor
import com.google.common.collect.Streams
import generations.gg.generations.core.generationscore.common.world.item.ItemWithLangTooltipImpl
import generations.gg.generations.core.generationscore.common.world.item.LangTooltip
import generations.gg.generations.core.generationscore.common.world.item.PostBattleUpdatingItem
import generations.gg.generations.core.generationscore.common.world.item.PostBattleUpdatingItem.BattleData
import net.minecraft.world.item.ItemStack

class SingleElmentPostUpdatingItem(properties: Properties, private val type: ElementalType) : ItemWithLangTooltipImpl(properties), PostBattleUpdatingItem, LangTooltip {
    override fun checkData(player: PlayerBattleActor, stack: ItemStack, pixelmonData: BattleData): Boolean {
        return pixelmonData.pokemon.types.any { elementalType: ElementalType -> elementalType == type }
    }
}
